contrib/aws/aws-sdk-go/aws/aws.go

```diff
@@ -147,3 +163,224 @@ func awsAgent(req *request.Request) string {
+		}
+		destName = parts[len(parts)-1]
+	}
+	return map[string]string{
+		destTag: destName,
+	}, nil
+}
+
+func dynamoDBTags(params interface{}) (map[string]string, error) {
+	var tableName string
+	switch input := params.(type) {
+	case *dynamodb.GetItemInput:
+		tableName = *input.TableName
+	case *dynamodb.PutItemInput:
+		tableName = *input.TableName
+	case *dynamodb.QueryInput:
+		tableName = *input.TableName
+	case *dynamodb.ScanInput:
+		tableName = *input.TableName
+	case *dynamodb.UpdateItemInput:
+		tableName = *input.TableName
+	default:
+		return nil, nil
+	}
+	return map[string]string{
+		tags.DynamoDBTableName: tableName,
+	}, nil
+}
+
+func kinesisTags(params interface{}) (map[string]string, error) {
+	var streamName string
+	switch input := params.(type) {
+	case *kinesis.PutRecordInput:
+		streamName = *input.StreamName
+	case *kinesis.PutRecordsInput:
+		streamName = *input.StreamName
+	case *kinesis.AddTagsToStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.RemoveTagsFromStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.CreateStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.DeleteStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.DescribeStreamInput:
+		streamName = *input.StreamName
+	case *kinesis.DescribeStreamSummaryInput:
+		streamName = *input.StreamName
+	case *kinesis.GetShardIteratorInput:
+		streamName = *input.StreamName
+	default:
+		return nil, nil
+	}
+	return map[string]string{
+		tags.KinesisStreamName: streamName,
+	}, nil
+}
+
+func eventBridgeTags(params interface{}) (map[string]string, error) {
+	var ruleName string
+	switch input := params.(type) {
+	case *eventbridge.PutRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.DescribeRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.DeleteRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.DisableRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.EnableRuleInput:
+		ruleName = *input.Name
+	case *eventbridge.PutTargetsInput:
+		ruleName = *input.Rule
+	case *eventbridge.RemoveTargetsInput:
+		ruleName = *input.Rule
+	default:
+		return nil, nil
+	}
+	return map[string]string{
+		tags.EventBridgeRuleName: ruleName,
+	}, nil
+}
+
+func sfnTags(params interface{}) (map[string]string, error) {
+	nameFromExecutionARN := func(arn string) string {
+		parts := strings.Split(arn, ":")
+		return parts[len(parts)-2]
+	}
+	var stateMachineName, stateMachineArn string
+	switch input := params.(type) {
+	case *sfn.CreateStateMachineInput:
+		stateMachineName = *input.Name
+	case *sfn.DescribeStateMachineInput:
+		stateMachineArn = *input.StateMachineArn
+	case *sfn.StartExecutionInput:
+		stateMachineArn = *input.StateMachineArn
+	case *sfn.StopExecutionInput:
+		if input.ExecutionArn != nil {
+			stateMachineName = nameFromExecutionARN(*input.ExecutionArn)
+		}
+	case *sfn.DescribeExecutionInput:
+		if input.ExecutionArn != nil {
+			stateMachineName = nameFromExecutionARN(*input.ExecutionArn)
+		}
```

Comment:
Why do we need to do a nil-check for these, but not for the other fields in this file?

Reply:
we want to grab the state machine name, and the other payloads are operations that work directly with the state machine (`CreateStateMachine`, etc.). In those, the `StateMachineArn` field is mandatory.

These operations however, are against the `Execution` object, which needs the `ExecutionARN` instead, which contains the `StateMachineARN` within.

I will refactor `nameFromExecutionARN` to address the  previous comment + checking `nil` there (which in theory it shouldn't be nil because the field is marked as mandatory, but since the field is a pointer we probably should check anyway).